### PR TITLE
CCIP-4593 Create rmnproxy setRMN changeset

### PR DESCRIPTION
**Pull request by AnieeG**

### CODE REVIEW

**Review by connorwstein — APPROVED** [review]

**Review by asoliman92 — APPROVED** [review]

File: deployment/ccip/changeset/cs_update_rmn_config.go [hunk 1]
```diff
@@ -11,13 +11,113 @@ import (
...
+			return fmt.Errorf("RMNRemote not found for chain %d", chain)
+		}
+		if chainState.RMNProxy == nil {
+			return fmt.Errorf("RMNProxy not found for chain %d", chain)
+		}
+	}
+	return nil
+}
+
+func SetRMNRemoteOnRMNProxy(e deployment.Environment, cfg SetRMNRemoteOnRMNProxyConfig) (deployment.ChangesetOutput, error) {
```
Comment (makramkd) [inline deployment/ccip/changeset/cs_update_rmn_config.go:47]: Naming nit (not a big fan of it though) `SetRMNRemoteOnRMNProxyChangeset` (getting really Java-ey 😢 )

**Review by dimkouv — APPROVED** [review]

File: deployment/ccip/changeset/cs_update_rmn_config.go [hunk 2]
```diff
@@ -11,13 +11,113 @@ import (
...
+	state, err := LoadOnchainState(e)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to load onchain state: %w", err)
+	}
+	if err := cfg.Validate(state); err != nil {
+		return deployment.ChangesetOutput{}, err
+	}
+	var timelockBatch []timelock.BatchChainOperation
+	multiSigs := make(map[uint64]*gethwrappers.ManyChainMultiSig)
+	timelocks := make(map[uint64]common.Address)
```
Comment (makramkd) [inline deployment/ccip/changeset/cs_update_rmn_config.go:57]: We do this so often I feel like we should just have a helper like `state.GetAllMCMSes()` and `state.GetAllTimelocks()` 😛

File: deployment/ccip/changeset/cs_add_chain_test.go [hunk 3]
```diff
@@ -30,6 +30,7 @@ import (
 )
 
 func TestAddChainInbound(t *testing.T) {
+	t.Skipf("Skipping test as it is running into timeout issues, move the test into integration in-memory tests")
```
Comment (makramkd) [inline deployment/ccip/changeset/cs_add_chain_test.go:33]: Can you create a ticket for this? Its weird that its timing out, it takes 90s on my laptop

File: deployment/environment/memory/node.go [hunk 4]
```diff
@@ -286,6 +286,8 @@ func CreateKeys(t *testing.T,
 		}
 		backend := chain.Client.(*Backend).Sim
 		fundAddress(t, chain.DeployerKey, transmitters[evmChainID], assets.Ether(1000).ToInt(), backend)
+		// in sim chains the send transactions are performed with 0x0 address as the sender
```
Comment (makramkd) [inline deployment/environment/memory/node.go:289]: Do you have a link to the docs where this is mentioned?

**Review by makramkd — APPROVED** [review]

File: deployment/ccip/changeset/cs_add_chain_test.go (hunk 3 above)
Comment (AnieeG) [reply deployment/ccip/changeset/cs_add_chain_test.go:33]: Yes created https://smartcontract-it.atlassian.net/browse/CCIP-4594

**Pull request merged**